optaplanner-website-root/content/learn/drl-to-constraint-streams-migration.adoc

```diff
@@ -0,0 +1,447 @@
+# Migrating OptaPlanner score DRL to Constraint Streams
+:jbake-description: Migrate deprecated Score DRL to Constraint Streams
+:toc:
+
+Ever since its introduction back in 2020,
+the https://www.optaplanner.org/docs/optaplanner/latest/constraint-streams/constraint-streams.html[Constraint Streams framework] ("CS") has seen great adoption and has been used to implement any problem you can think of.
+Due to this success, we have finally decided to deprecate its predecessor, the score DRL framework.
+
+## What does deprecating score DRL mean?
+
+Users of OptaPlanner 8 can continue using score DRL without any concern as we will continue maintaining score DRL support for the lifetime of OptaPlanner 8.
+However, OptaPlanner 9 will completely remove all support for score DRL.
+Users of score DRL are therefore encouraged to start planning their migration from score DRL to some other scoring mechanism, preferrably CS.
+This document intends to make such migration as easy as possible, showing real-world examples of rules and constraints found in https://www.optaplanner.org/docs/optaplanner/latest/use-cases-and-examples/examples-overview/examples-overview.html[OptaPlanner Examples].
+
+## Benefits of Constraint Streams
+
+CS is a modern full-featured API for writing OptaPlanner constraints, offering several benefits over score DRL:
+
+* Developers do not need to learn any new language, CS is plain Java.
+* Full IDE support incl. syntax highlighting and refactoring.
+* Comprehensive support for unit testing.
+* In certain use cases, CS performs considerably faster than score DRL.
+
+## Migrating to Constraint Streams, step by step
+
+CS and DRL are very similar in their approach and in fact both use Drools to execute the constraint logic.
+Because of this, migrating many parts of score DRL to CS is straight-forward, even mechanical.
+However, this guide will not help you if your score DRL uses the following constructs, as there is no direct mapping between DRL and CS in that case:
```

Comment:
active tense
```suggestion
However, this guide does not help you if your score DRL uses the following constructs, as there is no direct mapping between DRL and CS in that case:
```

I would write this less defensively, more friendly/positive. Something like:

```suggestion
There are a few constructs that are hard to migrate easily and depend on a case by case basis. These are not covered in this guide:
```